app/controllers/likes_controller.rb

```diff
@@ -1,8 +1,8 @@
 class LikesController < ApplicationController
   def save_like
-    @like = Like.new(post_id: params[:post_id], user_id: current_user.id)
     @post_id = params[:post_id]
-    existing_like = Like.find_by(post_id: params[:post_id], user_id: current_user.id)
+    @like = Like.new(post_id: @post_id, user: current_user)
```

Comment:
This method is doing too much
You want to separate these out to be one single action
Delete should go into Delete
Save should go into Create